src/enzyme_ad/jax/Implementations/HLODerivatives.td

```diff
@@ -1107,7 +1107,7 @@ def : HLOInactiveOp<"RngBitGeneratorOp">;
 def : HLODerivative<"RsqrtOp", (Op $x),
                     [
                       // (Select (FCmpUEQ $x, (ConstantFP<"0"> $x)), (ConstantFP<"0"> $x), (FDiv (DiffeRet), (FMul (ConstantFP<"2"> $x), (Call<(SameFunc), [ReadNone,NoUnwind]> $x))))
-                      (Div (DiffeRet), (Mul (HLOConstantFP<"2"> $x), (Mul $x, (Sqrt $x))))
+                      (Neg (Div (DiffeRet), (Mul (HLOConstantFP<"2"> $x), (Mul $x, (Sqrt $x)))))
```

Comment:
Minor but can you put the neg on the mul and not on the div? This will enable better constant propagation (since often times differrt will be 1 or 0 or a phi thereof so then it simplifies nicely)

Reply:
Also actually why not just make the constant 2 into -2?

Reply:
This also only changes one AD mode, presumably you’ll need to change the 2 to be a -2 in the line above as well